tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -224,13 +235,19 @@ const dataReducer: ActionReducer<RunsDataState, Action> = createReducer(
     const groupKeyToColorString = new Map(state.groupKeyToColorString);
     const defaultRunColorForGroupBy = new Map(state.defaultRunColorForGroupBy);
 
+    let groupby = state.initialGroupBy;
+    if (state.userSetGroupByKey !== null) {
+      groupby = createGroupBy(
+        state.userSetGroupByKey,
+        state.colorGroupRegexString
+      );
+    }
     const groups = groupRuns(
-      state.userSetGroupByKey
-        ? createGroupBy(state.userSetGroupByKey, state.colorGroupRegexString)
-        : state.initialGroupBy,
+      groupby,
```

Comment:
Just wondering, do you need this change? The code looks about the same to me.

Reply:
what was wrong is,
When `userSetGroupByKey = GroupByKey.RUN` it's `0` which would result in `groups` to be `state.initialGroupBy`
I mean to check null but what I wrote did not.